src/containerapp/azext_containerapp/custom.py

```diff
@@ -0,0 +1,954 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.azclierror import (RequiredArgumentMissingError, ResourceNotFoundError, ValidationError)
+from azure.cli.core.commands.client_factory import get_subscription_id
+from azure.cli.core.util import sdk_no_wait
+from knack.util import CLIError
+from knack.log import get_logger
+from msrestazure.tools import parse_resource_id, is_valid_resource_id
+from msrest.exceptions import DeserializationError
+
+from ._client_factory import handle_raw_exception
+from ._clients import ManagedEnvironmentClient, ContainerAppClient
+from ._sdk_models import *
+from ._models import (
+    ManagedEnvironment as ManagedEnvironmentModel,
+    VnetConfiguration as VnetConfigurationModel,
+    AppLogsConfiguration as AppLogsConfigurationModel,
+    LogAnalyticsConfiguration as LogAnalyticsConfigurationModel,
+    Ingress as IngressModel,
+    Configuration as ConfigurationModel,
+    Template as TemplateModel,
+    RegistryCredentials as RegistryCredentialsModel,
+    ContainerApp as ContainerAppModel,
+    Dapr as DaprModel,
+    ContainerResources as ContainerResourcesModel,
+    Scale as ScaleModel,
+    Container as ContainerModel)
+from ._utils import (_validate_subscription_registered, _get_location_from_resource_group, _ensure_location_allowed,
+                    parse_secret_flags, store_as_secret_and_return_secret_ref, parse_list_of_strings, parse_env_var_flags,
+                    _generate_log_analytics_if_not_provided, _get_existing_secrets, _convert_object_from_snake_to_camel_case,
+                    _object_to_dict, _add_or_update_secrets, _remove_additional_attributes, _remove_readonly_attributes,
+                    _add_or_update_env_vars, _add_or_update_tags, update_nested_dictionary, _add_or_update_traffic_Weights,
+                    _get_app_from_revision)
+
+logger = get_logger(__name__)
+
+
+# These properties should be under the "properties" attribute. Move the properties under "properties" attribute
+def process_loaded_yaml(yaml_containerapp):
+    if not yaml_containerapp.get('properties'):
+        yaml_containerapp['properties'] = {}
+
+    nested_properties = ["provisioningState", "managedEnvironmentId", "latestRevisionName", "latestRevisionFqdn", "customDomainVerificationId", "configuration", "template", "outboundIPAddresses"]
```

Comment:
Will some of these features like 'CustomDomainVerificationId' be ready for the initial release? if no can we remove these, esp. if this might cause some confusion or errors?

Reply:
This was added since so that once these features are available on the API side, the CLI wouldn't need an update to support it